File: slices.py
#!/usr/bin/env python3

def slices(n: int) -> int:
  acc, a, b = 0, 0, 1
  while a < n:
    s = (a + b) * (b - a + 1) / 2;

    if s == n:
      acc += 1
      b += 1
    elif s < n:
      b += 1
    else:
      a += 1

  return acc + 1

if __name__ == '__main__':
  best = 0
  n = 100000
  for i in range(n):
    ret = slices(i)
    if ret > best:
      best = ret
      print(f"found new best at: {i} = {ret}")

File: slices.c
#include <stdio.h>
#include <stddef.h>

int64_t slices(int64_t n) {
  int64_t a = 1, b = 2, acc = 0;
  while (a < n) {
    int64_t s = (a + b) * (b - a + 1) / 2;
    if (s == n) {
      acc++;
      b++;
    } else if (s < n) {
      b++;
    } else {
      a++;
    }
  }
  return acc + 1;
}

int main() {
  int32_t n = 100000;
  int32_t best = 0;
  for (int64_t i = 0; i < n; ++i) {
    int64_t ret = slices(i);
    if (ret > best) {
      printf("found new best at: %lld = %lld\n", i, ret);
      best = ret;
    }
  }
  return 0;
}
